app/Services/MenuService/SidebarMenuService.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+namespace App\Services\MenuService;
+
+use App\Services\MenuService\AdminMenuItem;
+
+class SidebarMenuService
+{
+    protected $groups = [];
+
+    /**
+     * Add a menu item to the sidebar.
+     *
+     * @param AdminMenuItem|array $item The menu item or configuration array
+     * @param string|null $group The group to add the item to
+     * @return void
+     * @throws \InvalidArgumentException
+     */
+    public function addMenuItem($item, $group = null)
+    {
+        // If item is an array, convert it to AdminMenuItem
+        if (is_array($item)) {
+            $menuItem = $this->createFromArray($item);
+        } else if ($item instanceof AdminMenuItem) {
+            $menuItem = $item;
+        } else {
+            throw new \InvalidArgumentException('MenuItem must be an array or AdminMenuItem instance');
+        }
+        
+        if (!isset($this->groups[$group])) {
+            $this->groups[$group] = [];
+        }
+        $this->groups[$group][] = $menuItem;
+    }
+
+    /**
+     * Create an AdminMenuItem from an array configuration.
+     *
+     * @param array $data Configuration data
+     * @return AdminMenuItem
+     */
+    protected function createFromArray(array $data): AdminMenuItem
+    {
+        $menuItem = new AdminMenuItem();
+        if (isset($data['children']) && is_array($data['children'])) {
+            $children = [];
+            foreach ($data['children'] as $child) {
+                $children[] = $this->createFromArray($child);
+            }
+            $data['children'] = $children;
+        }
+        
+        return $menuItem->Html($data);
```

Comment:
Need to change method name ?
`setAttributes()` maybe.

and variables coming should be `array $menuItems`